migrations/helpers/shared_helper.py

```diff
@@ -131,78 +116,94 @@ def get_cherrypicked_samples(
+
+    return build_data_frame_from_database_queries_aggregation(
+        database_connection_uri(config),
+        SQL_MLWH_GET_CP_SAMPLES_BY_DATE,
+        params_for_cherrypicked_samples_by_date_query(
+            root_sample_ids, plate_barcodes, start_date, end_date, chunk_size
+        ),
     )
 
 
-def __beckman_cherrypicked_samples_query(ml_wh_db: str, events_wh_db: str) -> str:
-    """Forms the SQL query to identify samples cherrypicked via the Beckman workflow.
+def remove_cherrypicked_samples(samples: List[SampleDoc], cherry_picked_samples: List[List[str]]) -> List[SampleDoc]:
+    """Remove samples that have been cherry-picked. We need to check on (root sample id, plate barcode) combo rather
+    than just root sample id. As multiple samples can exist with the same root sample id, with the potential for one
+    being cherry-picked, and one not.
 
-    Arguments:
-        ml_wh_db {str} -- The name of the MLWH database
-        events_wh_db {str} -- The name of the Events Warehouse database
+    Args:
+        samples (List[Sample]): List of samples in the shape of mongo documents
+        cherry_picked_samples (List[List[str]]): 2 dimensional list of cherry-picked samples with root sample id and
+        plate barcodes for each.
 
     Returns:
-        str -- the SQL query for Beckman cherrypicked samples
+        List[Sample]: The original list of samples minus the cherry-picked samples.
     """
-    return (
-        f"SELECT mlwh_sample.description AS `{FIELD_ROOT_SAMPLE_ID}`, mlwh_lh_sample.plate_barcode AS `{FIELD_PLATE_BARCODE}`,"  # noqa: E501
-        f" mlwh_sample.phenotype AS `Result_lower`, mlwh_lh_sample.coordinate AS `{FIELD_COORDINATE}`"  # noqa: E501
-        f" FROM {ml_wh_db}.sample AS mlwh_sample"
-        f" JOIN {ml_wh_db}.lighthouse_sample AS mlwh_lh_sample ON (mlwh_sample.uuid_sample_lims = mlwh_lh_sample.lh_sample_uuid)"  # noqa: E501
-        f" JOIN {events_wh_db}.subjects AS mlwh_events_subjects ON (mlwh_events_subjects.uuid = UNHEX(REPLACE(mlwh_lh_sample.lh_sample_uuid, '-', '')))"  # noqa: E501
-        f" JOIN {events_wh_db}.roles AS mlwh_events_roles ON (mlwh_events_roles.subject_id = mlwh_events_subjects.id)"  # noqa: E501
-        f" JOIN {events_wh_db}.events AS mlwh_events_events ON (mlwh_events_events.id = mlwh_events_roles.event_id)"  # noqa: E501
-        f" JOIN {events_wh_db}.event_types AS mlwh_events_event_types ON (mlwh_events_event_types.id = mlwh_events_events.event_type_id)"  # noqa: E501
-        f" WHERE mlwh_sample.description IN %(root_sample_ids)s"
-        f" AND mlwh_lh_sample.plate_barcode IN %(plate_barcodes)s"
-        f" AND mlwh_events_event_types.key = '{PLATE_EVENT_DESTINATION_CREATED}'"
-        " GROUP BY mlwh_sample.description, mlwh_lh_sample.plate_barcode, mlwh_sample.phenotype, mlwh_lh_sample.coordinate;"  # noqa: E501
+    cherry_picked_sets = [{cp_sample[0], cp_sample[1]} for cp_sample in cherry_picked_samples]
+    return list(
```

Comment:
I think first storing the filter as a variable then returning the list will be more readable.